In [1]:
import pandas as pd
from scipy import stats




## Importing Data

In [7]:
lipstick = pd.read_csv('/Users/britfathi/Downloads/lead_lipstick.csv')

In [8]:
lipstick.head()

JRC_code purchCntry prodCntry    Pb  sdPb shade prodType  priceCatgry
0     C135         NL        NL  3.75  0.24   Red       LP            2
1      C18         FI        FI  2.29  0.07   Red       LP            2
2      C20         FI        IT  1.27  0.06   Red       LP            2
3     C164         DE        FR  1.21  0.06   Red       LP            2
4      C71         MT        UK  0.85  0.04   Red       LP            2

In [10]:
lipstick.shade.unique()

array(['Red', 'Purple', 'Pink', 'Brown'], dtype=object)

## Analyzing if the lipstick shade influences the priceCatgry are related. 

In [11]:
lipstick_crosstab = pd.crosstab(lipstick['shade'], lipstick['priceCatgry'])

In [12]:
stats.chi2_contingency(lipstick_crosstab)

Chi2ContingencyResult(statistic=7.860569553614045, pvalue=0.2484973879479863, dof=6, expected_freq=array([[14.26008969, 36.32286996,  9.41704036],
       [19.25112108, 49.03587444, 12.71300448],
       [ 8.79372197, 22.39910314,  5.80717489],
       [10.69506726, 27.24215247,  7.06278027]]))

### This is an array of 4 rows representing the shades and 3 columns representing the priceCatgry. 
## because all values of the array are greater than 5 the assumption has been met (no relation)
### The Chi-Square stat is 7.86, the p value is 0.248. 
### Because the p value is > .05, there is not a significant relationship between lipstick shade and price category. 
### No shade of lipstick is pricier or cheaper than the other. 